src/main/java/net/snowflake/ingest/streaming/internal/ParquetChunkData.java

```diff
@@ -34,6 +35,15 @@ public ParquetChunkData(
     this.rows = rows;
     this.parquetWriter = parquetWriter;
     this.output = output;
-    this.metadata = metadata;
+    // create a defensive copy of the parameter map.
+    this.metadata = createDefensiveCopy(metadata);
```

Comment:
Should we use `Map.copyOf(metadata)` to create an immutable map to ensure no more modifcation per design?

Reply:
There is a problem because of the Java level that the SDK is using. Map.copyOf() we need level 10.